src/python/turicreate/toolkits/activity_classifier/_tf_model_architecture.py

```diff
@@ -41,219 +42,128 @@ def __init__(
 
-        # Optimizer
-        update_ops = _tf.get_collection(_tf.GraphKeys.UPDATE_OPS)
-        self.set_learning_rate(1e-3)
-        train_op = self.optimizer.minimize(self.loss_op)
-        self.train_op = _tf.group([train_op, update_ops])
+        # Final dense layer
+        dense3 = keras.layers.Dense(num_classes, use_bias=False)
+        cur_outputs = dense3(cur_outputs)
 
-        # Initialize all variables
-        self.sess.run(_tf.global_variables_initializer())
-        self.sess.run(_tf.local_variables_initializer())
+        # Softmax layer
+        softmax = keras.layers.Softmax()
+        cur_outputs = softmax(cur_outputs)
 
-        self._load_lstm_biases(net_params)
-
-    def __del__(self):
-        self.sess.close()
-        self.gpu_policy.stop()
-
-    def _get_lstm_weights_params(self, net_params):
-        """
-        Function to get LSTM weights from the C++ implementation in TensorFlow format
-
-        Parameters
-        ----------
-        net_params: Dictionary
-            Dict with weights from the C++ implementation and  its names
+        self.model = keras.Model(inputs=inputs, outputs=cur_outputs)
+        self.model.compile(
+            loss=tf.losses.categorical_crossentropy,
+            optimizer=keras.optimizers.Adam(learning_rate=1e-3),
+            sample_weight_mode="temporal"
+        )
 
-        Returns
-        -------
+        #############################################
+        # Load the Weights of the Neural Network
+        #############################################
+        for key in net_params.keys():
+            net_params[key] = _utils.convert_shared_float_array_to_numpy(net_params[key])
 
-        lstm: lstm weights in Tensorflow Format
-        """
-        i2h_i = net_params["lstm_i2h_i_weight"]
-        i2h_f = net_params["lstm_i2h_f_weight"]
-        i2h_c = net_params["lstm_i2h_c_weight"]
-        i2h_o = net_params["lstm_i2h_o_weight"]
-        h2h_i = net_params["lstm_h2h_i_weight"]
-        h2h_f = net_params["lstm_h2h_f_weight"]
-        h2h_c = net_params["lstm_h2h_c_weight"]
-        h2h_o = net_params["lstm_h2h_o_weight"]
-        lstm = _utils.convert_lstm_weight_coreml_to_tf(
-            i2h_i, i2h_c, i2h_f, i2h_o, h2h_i, h2h_c, h2h_f, h2h_o
+        # Set weight for first dense layer
+        l = self.model.layers[1]
+        l.set_weights(
+            (_utils.convert_conv1d_coreml_to_tf(net_params["conv_weight"]),
+             net_params["conv_bias"])
         )
-        return lstm
 
-    def _load_lstm_biases(self, net_params):
-        """
-        Function to load LSTM biases from the C++ implementation into TensorFlow
-
-        Parameters
-        ----------
-        net_params: Dictionary
-            Dict with weights from the C++ implementation and  its names
+        # Set LSTM weights
+        i2h, h2h, bias = [], [], []
+        for i in ('i', 'f', 'c', 'o'):
+            i2h.append(eval('net_params["lstm_i2h_%s_weight"]' % i))
```

Comment:
Can you make this logic a utility function like earlier?

Reply:
Sure, I'm happy to do that. This new function would only be called from one place, so I don't think it makes sense to put it in a separate file like it was before this change. If we want to put this code in a separate function, I would probably just add it as an inner function here, i.e. add this code as an inner function to this `__init__` method. I guess maybe that only really makes sense if it improves readability.

@shreyajain17 - what do you think?